google/reader.go

```diff
@@ -107,3 +132,69 @@ func (r *GCPReader) ListNetworks(ctx context.Context, filter string) ([]compute.
 	}
 	return networks, nil
 }
+
+// ListHealthCheck returns a list of health checks in the project
+func (r *GCPReader) ListHealthCheck(ctx context.Context, filter string) ([]compute.HealthCheck, error) {
+	ns := compute.NewHealthChecksService(r.compute)
+	// 500 is the current `maxResults`
+	checks := make([]compute.HealthCheck, 0, 500)
+	if err := ns.List(r.project).
+		Filter(filter).
+		Pages(ctx, func(list *compute.HealthCheckList) error {
+			for _, check := range list.Items {
+				checks = append(checks, *check)
+			}
+			return nil
+		}); err != nil {
+		return nil, err
+	}
+	return checks, nil
```

Comment:
I guess you could just do `err := ns.List` and then `return checks, err` so the `err != nil` check won't be necessary